cpp/include/cudf/strings/regex/flags.hpp

```diff
@@ -60,6 +61,17 @@ constexpr bool is_dotall(regex_flags const f)
   return (f & regex_flags::DOTALL) == regex_flags::DOTALL;
 }
 
+/**
+ * @brief Returns true if the given flags contain ASCII.
+ *
+ * @param f Regex flags to check
+ * @return true if `f` includes ASCII
+ */
+constexpr bool is_ascii(regex_flags const f)
+{
+  return (f & regex_flags::ASCII) == regex_flags::ASCII;
```

Comment:
I guess this is just copy-pasted from above so not a big deal, but I'm used to see bit field checks without the `==` sign. After all, all you really need to verify is that you have a nonzero result, assuming that the bit fields are properly distinct.

Reply:
That is true. I was trying to make this a strict boolean operation to make it clear what is happening. 
The optimizer generates the same code either way -- the `==` is removed.

Reply:
Agreed, it doesn't matter for performance and is probably clearer to readers who aren't already used to playing with bit flags.